test/server/proxy-test.js

```diff
@@ -1669,98 +1673,116 @@ describe('Proxy', () => {
 
             const requestTime = Date.now();
 
-            request(options, err => {
-                const responseTime = Date.now();
+            return request(options)
+                .then(() => {
+                    expect.fail('Request should raise an error');
+                })
+                .catch(err => {
+                    const responseTime = Date.now();
 
-                expect(err.toString()).include('socket hang up');
-                expect(responseTime - requestTime).above(DestinationRequest.XHR_TIMEOUT);
+                    expect(err.toString()).include('socket hang up');
+                    expect(responseTime - requestTime).above(DestinationRequest.XHR_TIMEOUT);
 
-                DestinationRequest.TIMEOUT     = savedReqTimeout;
-                DestinationRequest.XHR_TIMEOUT = savedXhrReqTimeout;
-                done();
-            });
+                    DestinationRequest.TIMEOUT     = savedReqTimeout;
+                    DestinationRequest.XHR_TIMEOUT = savedXhrReqTimeout;
+                });
         });
 
+        // // Old 'request'-module version
```

Comment:
remove comment and test.
Now (check with `6.12.0`) `nodejs` is raised errro for this case:
```js
TypeError: The header content contains invalid characters
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:359:11)
    at ServerResponse.res.setHeader (node_modules\connect\lib\patch.js:59:22)
    at ServerResponse.writeHead (_http_server.js:180:21)
    at ServerResponse.res.writeHead (node_modules\connect\lib\patch.js:75:22)
```